File: assignments/assignment-1-solution/src-csharp/CrazyEights/Domain/Suit.cs
using System.Collections.Immutable;

namespace CrazyEights.Domain;

/// <summary>
/// Provides helpers for suit names, symbols, and lists.
/// </summary>
public static class Suit
{
    public static string GetSuitSymbol(SuitType suit)
    {
        return suit switch
        {
            SuitType.Hearts => "♥",
            SuitType.Diamonds => "♦",
            SuitType.Clubs => "♣",
            SuitType.Spades => "♠",
            _ => throw new ArgumentOutOfRangeException(nameof(suit), suit, null)
        };
    }

    public static string GetSuitName(SuitType suit)
    {
        return $"{suit} {GetSuitSymbol(suit)}";
    }

    public static IReadOnlyList<SuitType> GetSuits()
    {
        return Enum.GetValues<SuitType>()
            .Where(suit => suit != SuitType.NotSet)
            .ToImmutableList();
    }
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/domain/Suit.java
package edu.kennesaw.crazy8s.domain;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Provides helpers for suit names, symbols, and lists.
 */
public final class Suit {
    private Suit() {
    }

    public static String getSuitSymbol(SuitType suit) {
        switch (suit) {
            case HEARTS:
                return "♥";
            case DIAMONDS:
                return "♦";
            case CLUBS:
                return "♣";
            case SPADES:
                return "♠";
            default:
                throw new IllegalArgumentException("Unknown suit: " + suit);
        }
    }

    public static String getSuitName(SuitType suit) {
        return String.format("%s %s", suit.getDisplayName(), getSuitSymbol(suit));
    }

    public static List<SuitType> getSuits() {
        List<SuitType> suits = new ArrayList<>();
        for (SuitType suit : SuitType.values()) {
            if (suit != SuitType.NOT_SET) {
                suits.add(suit);
            }
        }

        return Collections.unmodifiableList(suits);
    }
}
